plugin/trino-snowflake/src/test/java/io/trino/plugin/snowflake/TestSnowflakeConnectorTest.java

```diff
@@ -367,4 +367,35 @@ public void testSelectInformationSchemaColumns()
     {
         // TODO https://github.com/trinodb/trino/issues/21157 Enable this test after fixing the timeout issue
     }
+
+    @Override
+    protected Optional<SetColumnTypeSetup> filterSetColumnTypesDataProvider(SetColumnTypeSetup setup)
+    {
+        switch (setup.newColumnType()) {
+            case "smallint":
+            case "integer":
+                return Optional.empty();
+        }
```

Comment:
Please leave a code comment why `setup.asUnsupported()` doesn't work.

Reply:
> INT , INTEGER , BIGINT , SMALLINT , TINYINT , BYTEINT are synonymous with NUMBER

I still couldn't understand why `SET DATA TYPE` doesn't work for these types. What's the error message when changing `bigint` to `integer`? Why can't we use `verifySetColumnTypeFailurePermissible`?

Reply:
All the number types are map to `NUMBER` in Snowflake, thus we can not determine the type comes from Snowflake, from client side we only know the column type is `BIGINT` with the different `columnSize`.

Snowflake doesn't return error message for the operation, it actual change the `columnSize`, i.e, change `tinyint -> int` corresponding to change `NUMBER(3,0)` -> `NUMBER(10, 0)`.

Reply:
>  it actual change the columnSize

So, it means we shouldn't skip this test.

Reply:
Then what's the expectation here, the set operation would be considered failed in Trino